MetOs-UiO/CTSM
# DOM

How to test with ctsm_dev_env_template:

1. Clone this repo: https://github.com/MetOs-UiO/ctsm_dev_env_template
2. Create a `.env` file in the repo root with the following content:
  ```
  CTSM_REPO=https://github.com/MetOs-UiO/CTSM.git
  CTSM_TAG=kaveh/dom
  INPUTDATA=/mnt/fram/noresm/inputdata
  ```
3. Download the prepared ALP1 data from here: https://ns2806k.webs.sigma2.no/EMERALD/EMERALD_platform/ALP1.tar
4. Extract the data in `data` folder in the cloned repo
5. Run `./run_docker.sh` in the repo root, then run the following commands in the opened shell within docker container (keep the shell open after running the commands for later steps):
  - `/ctsm-api/scripts/setup_case.sh -n ALP1 -c 
2000_DATM%GSWP3v1_CLM51%BGC_SICE_SOCN_SGLC_SWAV`
  - `cd /ctsm-api/resources/cases/ALP1`
  - `./xmlchange STOP_OPTION=nyears`
  - `export CESMDATAROOT=/ctsm-api/resources/data/ALP1`
6. Add the following to `/ctsm-api/resources/cases/ALP1/user_nl_clm` from within the docker container, or `./cases/ALP1/user_nl_clm` from the repo root (if it's not already there):
  ```
  paramfile = '$CLM_USRDAT_DIR/clm50_params.c211112.nc'

 do_harvest = .false.
 check_dynpft_consistency = .false.
  ```
7. In the shell you started in step 5, run `./case.build` and then `case.submit`. If these are successful, you should see output in `ctsm-api/resources/cases/ALP1/archive/lnd/hist` from within the container or in `cases/ALP1/archive/lnd/hist` from the repo folder.
